sdk/servicebus/Microsoft.Azure.ServiceBus/tests/SenderReceiverTests.cs

```diff
@@ -45,6 +45,30 @@ public async Task MessageReceiverAndMessageSenderCreationWorksAsExpected(bool pa
             });
         }
 
+        [Fact]
+        [LiveTest]
+        [DisplayTestMethodName]
+        public async Task ReceiveLotOfMessagesWithoutSettling()
+        {
+            // 5000 is the default limit on amqp session incoming window
+            int messageCount = 5010;
+            await ServiceBusScope.UsingQueueAsync(false, false, async queueName =>
+            {
+                var sender = new MessageSender(TestUtility.NamespaceConnectionString, queueName);
+                var receiver = new MessageReceiver(TestUtility.NamespaceConnectionString, queueName, receiveMode: ReceiveMode.PeekLock);
+
+                try
+                {
+                    await this.ReceiveDeleteTestCase(sender, receiver, messageCount);
```

Comment:
nit: Maybe we can call it `ReceiveNotSettlingTestCase`? Since we didn't pass the `ReceiveAndDelete` mode and by default receive is using PeekLock?